packages/inspection-report/src/components/common/Thumbnail.js

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Dimensions, Image, Text, StyleSheet, TouchableOpacity, View } from 'react-native';
+import { MaterialIcons } from '@expo/vector-icons';
+
+const styles = StyleSheet.create({
+  noImageWrapper: {
+    alignItems: 'center',
+    backgroundColor: '#f5f5f5',
+    borderRadius: 4,
+    justifyContent: 'center',
+  },
+  text: {
+    color: '#fff',
+    fontSize: 14,
+    marginTop: 5,
+    textAlign: 'center',
+  },
+});
+
+function Thumbnail({ image, click }) {
+  const [imageSize, setImageSize] = useState(109);
+  const { width } = Dimensions.get('window');
+
+  useEffect(() => {
+    if (width < 480) {
+      setImageSize(109);
+    } else if (width > 480 && width < 768) {
+      setImageSize(140);
+    } else {
+      setImageSize(175);
+    }
+  }, [width]);
```

Comment:
instead of doing this, you can remove the state and the useeffect and do something like : 

```javascript
// Declare this const OUTSIDE of the component, next to the styles
const responsiveImageSizes = [
  { breakpoint: 480, width: 109 },
  { breakpoint: 768, width: 140 },
  { breakpoint: null, width: 109 },
];

// Inside the component
const { width } = useWindowDimensions();
const imageSize = useMemo(
  () => responsiveImageSizes.find((r) => (!r.breakpoint || width <= r.breakpoint)).width,
  [width],
);
```

Reply:
Resolved all the comments.